### PR TITLE
Update the developer policy information on "Stay Informed" to refer to GitHub teams

### PR DESCRIPTION
**Pull request by joker-eph**

### CODE REVIEW

File: llvm/docs/DeveloperPolicy.rst [hunk 1]
```diff
@@ -46,21 +46,33 @@ quality.
 Stay Informed
 -------------
 
-Developers should stay informed by reading the `LLVM Discourse forums`_.
-If you are doing anything more than just casual work on LLVM, it is suggested that you also
-subscribe to the "commits" mailing list for the subproject you're interested in,
+Developers should stay informed by reading the `LLVM Discourse forums`_ and subscribing
+to the categories of interested for notifications.
```
Comment (pogo59) [inline llvm/docs/DeveloperPolicy.rst:50]: s/interested/interest/

File: llvm/docs/DeveloperPolicy.rst [hunk 2]
```diff
@@ -46,21 +46,33 @@ quality.
...
+
+Paying attention to changes being made by others is a good way to see what other people
+are interested in and watching the flow of the project as a whole.
+
+Contibution to the project are made through :ref:`GitHub Pull Requests <github-reviews>`
```
Comment (pogo59) [inline llvm/docs/DeveloperPolicy.rst:55]: Contributions

File: llvm/docs/DeveloperPolicy.rst [hunk 3]
```diff
@@ -46,21 +46,33 @@ quality.
...
+documentation. You can subscribe to notification for areas of the codebase by joining
```
Comment (pogo59) [inline llvm/docs/DeveloperPolicy.rst:56]: s/documentation//.  "Contributions are made through documentation" doesn't read right.

**Comment by joker-eph** [comment]
Thanks @pogo59 ; I fix the issues you found!

**Review by pogo59 — APPROVED** [review]
LGTM. But you might give it a bit more time to see if there are other comments.

**Comment by joker-eph** [comment]
@tstellar @tru : ping on this documentation udpate?

File: llvm/docs/DeveloperPolicy.rst [hunk 4]
```diff
@@ -46,21 +46,33 @@ quality.
...
-subscribe to the "commits" mailing list for the subproject you're interested in,
+Developers should stay informed by reading the `LLVM Discourse forums`_ and subscribing
+to the categories of interest for notifications.
+
+Paying attention to changes being made by others is a good way to see what other people
+are interested in and watching the flow of the project as a whole.
+
+Contibutions to the project are made through :ref:`GitHub Pull Requests <github-reviews>`.
+You can subscribe to notification for areas of the codebase by joining
+one of the `pr-subscribers-* <https://github.com/orgs/llvm/teams?query=issue-subscribers>`_
```
Comment (tstellar) [inline llvm/docs/DeveloperPolicy.rst:57]: This query should be pr-subscribers, not issue-subscribers.

**Review by tstellar — APPROVED** [review]

File: llvm/docs/DeveloperPolicy.rst (hunk 4 above)
Comment (joker-eph) [reply llvm/docs/DeveloperPolicy.rst]: Well spotted! That was a copy-paste from below...

**Pull request merged**